In [2]:
import pandas as pd
import numpy as np

from scipy.stats import ttest_ind
from scipy.stats import chi2_contingency
from scipy.stats import f_oneway

In [3]:
import pandas as pd

df = pd.read_csv("heart_disease_data.csv")

print(df.head())

   age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   48    1   0       130   256    1        0      150      1      0.0      2   
1   61    1   0       148   203    0        1      161      0      0.0      2   
2   44    0   2       118   242    0        1      149      0      0.3      1   
3   47    1   0       110   275    0        0      118      1      1.0      1   
4   56    1   3       120   193    0        0      162      0      1.9      1   

   ca  thal  target  
0   2     3       0  
1   1     3       0  
2   1     2       1  
3   1     2       0  
4   0     3       1  


# Descriptive Statistics

In [2]:
df.mean(numeric_only=True)

age          54.366337
sex           0.683168
cp            0.966997
trestbps    131.623762
chol        246.264026
fbs           0.148515
restecg       0.528053
thalach     149.646865
exang         0.326733
oldpeak       1.039604
slope         1.399340
ca            0.729373
thal          2.313531
target        0.544554
dtype: float64

# Median

In [3]:
df.median(numeric_only=True)

age          55.0
sex           1.0
cp            1.0
trestbps    130.0
chol        240.0
fbs           0.0
restecg       1.0
thalach     153.0
exang         0.0
oldpeak       0.8
slope         1.0
ca            0.0
thal          2.0
target        1.0
dtype: float64

# standard deviation

In [4]:
df.std(numeric_only=True)

age          9.074592
sex          0.465626
cp           1.031199
trestbps    17.523642
chol        51.787898
fbs          0.355903
restecg      0.525425
thalach     22.886223
exang        0.469406
oldpeak      1.160115
slope        0.615717
ca           1.021761
thal         0.611770
target       0.498422
dtype: float64

# Test 1: Independent T-Test

In [6]:
disease = df[df['target']==1]['chol']

no_disease = df[df['target']==0]['chol']

t_stat, p_value = ttest_ind(
    disease,
    no_disease
)

print("T Statistic:", t_stat)
print("P Value:", p_value)

T Statistic: -2.1025234021347012
P Value: 0.03592073156896611


# Null Hypothesis (H₀)

## There is no significant difference in cholesterol levels.

In [7]:
from scipy.stats import ttest_ind

print("Imported Successfully")

Imported Successfully


# The Independent T-Test was used to compare cholesterol levels between patients with and without heart disease. A p-value less than 0.05 indicates a statistically significant difference.

# Test 2: Chi-Square Test

# Null Hypothesis (H₀)

## Gender and heart disease are independent.

# Alternative Hypothesis (H₁)

### Gender and heart disease are associated.

# Create Contingency Table

In [9]:
table = pd.crosstab(
    df['sex'],
    df['target']
)

table

target,0,1
sex,,
0,48,144
1,228,186


# Perform Test

In [10]:
chi2, p, dof, expected = chi2_contingency(table)

print("Chi Square:", chi2)
print("P Value:", p)

Chi Square: 46.623926148596084
P Value: 8.600373213630996e-12


# Interpretation

In [11]:
if p < 0.05:
    print("Significant association")
else:
    print("No significant association")

Significant association


# Interview Explantion

## Chi-Square Test was used to determine whether gender and heart disease occurrence were statistically related.

# Test 3: ANOVA

# Null Hypothesis (H₀)

* Mean age is same across all chest pain groups.

# Alternative Hypothesis (H₁)

* At least one group has a different mean age.

In [12]:
group1 = df[df['cp']==0]['age']
group2 = df[df['cp']==1]['age']
group3 = df[df['cp']==2]['age']
group4 = df[df['cp']==3]['age']

In [13]:
f_stat, p_value = f_oneway(
    group1,
    group2,
    group3,
    group4
)

print("F Statistic:", f_stat)
print("P Value:", p_value)

F Statistic: 6.813367863429748
P Value: 0.0001604940260721571


# Interpretation

In [14]:
if p_value < 0.05:
    print("Significant difference")
else:
    print("No significant difference")

Significant difference


ANOVA was used to compare average age across multiple chest pain categories.

# 5. Confidence Interval

#Estimate population cholesterol range with 95% confidence.

In [4]:
mean = df['chol'].mean()

std = df['chol'].std()

n = len(df)

margin = 1.96 * (std / np.sqrt(n))

lower = mean - margin

upper = mean + margin

print("95% Confidence Interval")
print(lower, upper)

95% Confidence Interval
242.14069688765602 250.38735591762452


# INterpretation 

#We are 95% confident that the true average cholesterol level lies between the calculated lower and upper bounds.

# Correlation Analysis

In [5]:
corr = df.corr()['target']

corr.sort_values(
    ascending=False
)

target      1.000000
cp          0.433798
thalach     0.421741
slope       0.345877
restecg     0.137230
fbs        -0.028046
chol       -0.085239
trestbps   -0.144931
age        -0.225439
sex        -0.280937
thal       -0.344029
ca         -0.391724
oldpeak    -0.430696
exang      -0.436757
Name: target, dtype: float64

# Interpretation

Positive Correlation:

cp
thalach
slope

# negative correlation

oldpeak
exang
ca

# Statistical Findings Summary

Statistical Findings

1. Independent T-Test was used to compare cholesterol levels between patients with and without heart disease.

2. Chi-Square Test was conducted to evaluate the relationship between gender and disease occurrence.

3. ANOVA was applied to compare age across chest pain categories.

4. Confidence Intervals were calculated to estimate population cholesterol values.

5. Correlation Analysis identified the strongest predictors of heart disease.

6. Statistical testing validated relationships between clinical variables and disease risk.

# OUTPUT Report 

# Statistical Analysis Report

## Objective

The objective of this analysis was to statistically evaluate the relationship between clinical features and heart disease occurrence.

## Tests Performed

### 1. Independent T-Test

A T-Test was conducted to compare cholesterol levels between patients diagnosed with heart disease and patients without heart disease.

**Result:**

* T Statistic: **-2.1025**
* P Value: **0.0359**

**Conclusion:**
Since the p-value (0.0359) is less than 0.05, the null hypothesis is rejected. This indicates that cholesterol levels differ significantly between patients with heart disease and those without heart disease. Cholesterol appears to be an important clinical factor associated with heart disease risk.

---

### 2. Chi-Square Test

A Chi-Square Test was performed to examine the association between gender and heart disease occurrence.

**Result:**

* Chi-Square Statistic: **46.6239**
* P Value: **8.60 × 10⁻¹²**

**Conclusion:**
Since the p-value is significantly lower than 0.05, the null hypothesis is rejected. There is a statistically significant association between gender and heart disease occurrence. Gender may therefore be considered an important demographic factor in heart disease prediction.

---

### 3. ANOVA Test

ANOVA was applied to compare age distributions across different chest pain categories.

**Result:**

* F Statistic: **6.8134**
* P Value: **0.000160**

**Conclusion:**
Since the p-value is less than 0.05, the null hypothesis is rejected. This indicates that the average age differs significantly across chest pain categories. Age may influence the type or severity of chest pain experienced by patients.

---

## Correlation Analysis

Correlation analysis was performed to identify the most influential variables affecting heart disease prediction. Features with strong positive or negative correlations with the target variable were considered important predictors for subsequent machine learning model development.

---

## Summary

* Statistical testing confirmed that cholesterol levels significantly differ between patients with and without heart disease.
* A strong statistical association was observed between gender and heart disease occurrence.
* Significant age differences were found across chest pain categories.
* The findings validate the relevance of clinical and demographic variables for heart disease prediction.
* The results of this analysis support feature selection and predictive modeling in the machine learning phase.

## Business Insight

The statistical analysis suggests that cholesterol, gender, age, and chest pain characteristics play a significant role in heart disease risk assessment. These findings can assist healthcare professionals in identifying high-risk patients and support data-driven clinical decision-making.


In [8]:
df.columns

Index(['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach',
       'exang', 'oldpeak', 'slope', 'ca', 'thal', 'target'],
      dtype='object')